src/index.d.ts

```diff
@@ -275,9 +275,11 @@ export namespace h {
 // Preact render
 // -----------------------------------
 
+type Parent = Omit<Pick<Element | Document | ShadowRoot | DocumentFragment, 'nodeType' | 'parentNode' | 'firstChild' | 'insertBefore' | 'appendChild' | 'removeChild'>, 'childNodes'> & { childNodes: ArrayLike<Node> }
```

Comment:
Would it work if instead you did something like:
```ts
interface ContainerNode {
  nodeType: Node['nodeType'];
  parentNode: Node['nodeType'];
  firstChild: Node['firstchild'];
  insertBefore: Node['insertBefore'];
  appendChild: Node['appendChild'];
  removeChild: Node['removeChild'];
  childNodes: ArrayLike<Node>
}
```

I find this a bit easier to read and my understanding is that all the types (`Element | Document | ShadowRoot | .. etc`) already implement this interface so it should work?

Reply:
I just checked on this and you are correct that all of these properties come from `Node`, so what you have proposed would work.  I'll make this change to the PR.

Reply:
Applied the change.  Let me know if there are other things I should change!